src/recorder.js

```diff
@@ -124,13 +124,31 @@ Recorder.prototype.initSourceNode = function( sourceNode ){
 
 Recorder.prototype.initWorker = function(){
   var self = this;
-  var streamPage = function( e ) { self.streamPage( e.data ); };
-  var storePage = function( e ) { self.storePage( e.data ); };
+  var onPage = (this.config.streamPages ? this.streamPage : this.storePage).bind(this);
 
   this.recordedPages = [];
   this.totalLength = 0;
   this.encoder = new global.Worker( this.config.encoderPath );
-  this.encoder.addEventListener( "message", this.config.streamPages ? streamPage : storePage );
+  return new Promise((resolve, reject) => {
+    this.encoder.addEventListener( "message", function(e) {
+      switch( e['data']['message'] ){
+        case 'ready':
+          resolve();
+          break;
+        case 'page':
+          onPage(e['data']['page']);
+          break;
+        case 'done':
+          this.finish();
+          break;
+      }
+    });
+    this.encoder.postMessage( Object.assign({
```

Comment:
This will never be reached after the return statement.  Should this block go inside the `new Promise`?

Reply:
Never mind. I did not read the code correctly.